generator/docs/GeneratorSetup.md

```diff
@@ -0,0 +1,142 @@
+This is only necessary when your Maven dependencies change. Otherwise prefer the IDE's built-in Build.
+
+Navigate to the `App.java` file (...\data-engineering-generator\generator\src\main\java\com\scottlogic\deg\generator\App.java). Right click and debug - *this will fail*.
+
+Now edit the run configuration on the top toolbar created by the initial run. Name the run configuration 'Generate' and under 'Program Arguments' enter the following, replacing the paths with your desired locations:
+
+```
+generate "<path to an example JSON profile>" "<path to desired output CSV>"
+```
+
+Additionally create another run configuration called GenerateTestCases and add the program arguments
+
+```
+generateTestCases "<path to an example JSON profile>" "<path to desired output folder for generated CSVs>"
+```
+
+Run both of these configurations to test that installation is successful.
+
+### Eclipse
+
+You can set up an existing Eclipse to support Scala or set up a new Eclipse with Scala support
+
+#### Set up an existing Eclipse installation
+
+Your Eclipse will need the `scala-ide` plug-in and the `m2e-scala` connector
+
+##### `scala-ide`
+
+You can follow the installation instruction on 
+[here](http://scala-ide.org/docs/current-user-doc/gettingstarted/index.html)
+
+As of this writing, we are using `Scala IDE for Eclipse` version 
+4.7.1.v-2_12-20180102121323-2dfe808 and `Scala 2.12.3` version 4.7.1.201801011322 (Note: 
+this version also supports Scala 2.11, the version that Spark needs).  Other versions may also work. 
+
+##### `m2e-scala` connector
+
+Adding the following url as a "Repository" in Eclipse (Help > Install New Software... > Add...)
+
+`http://alchim31.free.fr/m2e-scala/update-site/`
+
+And from there you install `Maven Integration for Scala IDE` (we use version 0.5.1.201410131304)
+
+#### Set up a new Eclipse installation with Scala support
+
+Get the [scala-ide](http://downloads.typesafe.com/scalaide-pack/4.7.0-vfinal-oxygen-212-20170929/scala-SDK-4.7.0-vfinal-2.12-win32.win32.x86_64.zip)
+version of Eclipse.
+
+Extract the archive, launch Eclipse and choose your workspace location.
+
+#### Import the project as a Maven project.
+
+For example, in Eclipse, File > Import... > Maven > Existing Maven Project, and then point to the 
+top-level directory of this project `data-engineering-generator`
+
+When importing is complete some errors relating to Scala versions may be shown.
+To fix this:
+
+- Right-click the *profiler* project in the Package Explorer view and click
+    *Properties*
+ - Select *Scala Compiler*, tick *Use Project Settings*, then select *Latest
+    2.11 bundle*
+ - Click *Apply and Close*
+
+After the project has been rebuilt the errors should be gone.
+
+#### Run
+
+The appropriate run configuration is stored in the repository and so can be run
+straight away:
+
+- In the top toolbar, click the arrow next to the green start button
+- Click *Profiler*

```

Comment:
Shouldn't mention the `profiler`, should be `generator`

Reply:
This is my laziness - since I didn't use eclipse to install and get things working I didn't read this through properly. Since we all seem to be using IntelliJ do we need to keep Eclipse instructions?

Reply:
Removed eclipse instructions for now as they are not relevant here.